lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -669,6 +671,17 @@ private function determineIdGeneratorStrategy(AbstractPlatform $platform): int
         return ClassMetadata::GENERATOR_TYPE_TABLE;
     }
 
+    private function truncateSequenceName(string $schemaElementName): string
+    {
+        $maxIdentifierLength = $this->getTargetPlatform()->getMaxIdentifierLength();
+
+        if (strlen($schemaElementName) > $maxIdentifierLength) {
+            return substr($schemaElementName, 0, $maxIdentifierLength);
+        }
+
+        return $schemaElementName;
+    }
```

Comment:
It's weird because although all platforms have a max identifier length since they inherit it from `AbstractPlatform`, only SQLAnywhere and Oracle since to act on it… for other platforms, I suppose there is a crash and that's it.

Maybe we should trigger a deprecation on line 679 saying this will happen in the future? Or maybe even change the condition so that it only applies for `sqlanywhere` and `oracle`?

Reply:
In any case, those sequence names that we truncate here are generated by our own code, no user input. A deprecation would be confusing, I guess.

I've limited the truncation to those two platforms now.